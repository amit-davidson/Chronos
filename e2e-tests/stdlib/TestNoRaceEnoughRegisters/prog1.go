package main

func main() {
	// from erf.go
	const (
		sa1 = 1
		sa2 = 2
		sa3 = 3
		sa4 = 4
		sa5 = 5
		sa6 = 6
		sa7 = 7
		sa8 = 8
	)
	var s, S float64
	s = 3.1415
	S = 1 + s*(sa1+s*(sa2+s*(sa3+s*(sa4+s*(sa5+s*(sa6+s*(sa7+s*sa8)))))))
	s = S
}
