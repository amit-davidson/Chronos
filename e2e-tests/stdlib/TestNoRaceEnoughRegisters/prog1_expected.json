{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:60",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:60",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:58",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:58",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:60",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:53",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 5
				}
			}
		},
		{
			"ID": 6,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:53",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:58",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 7,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:51",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 8,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:51",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:53",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		},
		{
			"ID": 9,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:46",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				}
			}
		},
		{
			"ID": 10,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:46",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:51",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 10
				}
			}
		},
		{
			"ID": 11,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:44",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 11
				}
			}
		},
		{
			"ID": 12,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:44",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:46",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 12
				}
			}
		},
		{
			"ID": 13,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:39",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 13
				}
			}
		},
		{
			"ID": 14,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:39",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:44",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 14
				}
			}
		},
		{
			"ID": 15,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:37",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 15
				}
			}
		},
		{
			"ID": 16,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:37",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:39",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 16
				}
			}
		},
		{
			"ID": 17,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:32",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 17
				}
			}
		},
		{
			"ID": 18,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:32",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:37",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 18
				}
			}
		},
		{
			"ID": 19,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:30",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 19
				}
			}
		},
		{
			"ID": 20,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:30",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:32",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 20
				}
			}
		},
		{
			"ID": 21,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:25",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 21
				}
			}
		},
		{
			"ID": 22,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:25",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:30",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 22
				}
			}
		},
		{
			"ID": 23,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:23",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 23
				}
			}
		},
		{
			"ID": 24,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:23",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:25",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 24
				}
			}
		},
		{
			"ID": 25,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:18",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 25
				}
			}
		},
		{
			"ID": 26,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:18",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:23",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 26
				}
			}
		},
		{
			"ID": 27,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:16",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 27
				}
			}
		},
		{
			"ID": 28,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:16",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:18",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 28
				}
			}
		},
		{
			"ID": 29,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:11",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 29
				}
			}
		},
		{
			"ID": 30,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:11",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:16",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 30
				}
			}
		},
		{
			"ID": 31,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:8",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 31
				}
			}
		},
		{
			"ID": 32,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:8",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceEnoughRegisters/prog1.go:17:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 32
				}
			}
		}
	]
}